typings/index.d.ts

```diff
@@ -1183,6 +1183,34 @@ declare module 'discord.js' {
     public isSelectMenu(): this is SelectMenuInteraction;
   }
 
+  export class InteractionCollector<T extends Interaction> extends Collector<Snowflake, T> {
+    constructor(client: Client, options?: InteractionCollectorOptions<T>);
+    private _handleMessageDeletion(message: Message): void;
+    private _handleChannelDeletion(channel: GuildChannel): void;
+    private _handleGuildDeletion(guild: Guild): void;
+
+    public channel: TextChannel | NewsChannel | DMChannel | null;
+    public componentType: MessageComponentType | null;
+    public empty(): void;
+    public readonly endReason: string | null;
+    public guild: Guild | null;
+    public interactionType: InteractionType | null;
+    public message: Message | null;
+    public options: InteractionCollectorOptions<T>;
+    public total: number;
+    public users: Collection<Snowflake, User>;
+
+    public collect(interaction: Interaction): Snowflake;
+    public dispose(interaction: Interaction): Snowflake;
+    public on(event: 'collect' | 'dispose', listener: (interaction: T) => Awaited<void>): this;
```

Comment:
```suggestion
    public on(event: 'collect' | 'dispose', listener: (interaction: T) => void): this;
```
void works just the same, no real reason to use `Awaited<void>` over just `void` ?

Reply:
It's the same as `Client#on()`, which uses Awaited too

Reply:
Pretty sure the reason it was switched to `Awaited<void>` is because otherwise you can't have async listeners.

Reply:
Yeah just found #5602

Reply:
didn't get any errors on [ts playground](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAZgVzMGBeGAKOYBcmAeeARiCADYCmAhmAJRoB8MAbiAJYAm9qT2GUAJwQVaAbgBQ4xMgxUIATxncmAb3EwNMARSgIBYGACJoAtmADmhiQF8xk6cAwZlMNZq069BsBQDuMAAUBEABbNggKAB4TM3MGDG0IcmYKRg8kshSMY0FYw1o7W1EgA) 🤷 strange

Reply:
> didn't get any errors on [ts playground](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAZgVzMAXDAFHMb0A80BGIIANgKYCGYAlDALwB8MAbiAJYAmtjL7H9GLDnwwipSjXpNWnbkyzooAJwRlqAbgBQmxMnQUIATz1yYAb00wrMJWSgIlYGACJoStmADmzrQF8N2rrA6OimFtY2dg5OYGQA7jAACkogALZsEGQAPG4engzothCkzGRSkcUkpeiuynnO1AH+6kA)  strange

From the pull request @DTrombett linked
> When using `client.on` with a callback that returns a `Promise`, eslint will then complain if the `no-misused-promises` (from `@typescript-eslint` and various plugins) is enabled, as `Promise<?>` does not match `void`.

Reply:
ah, makes sense